utils/setupsOnTheGo.ts

```diff
@@ -1312,3 +1314,87 @@ export async function replaceByStateCall(
   const parsed = api.createType(returnType, res);
   console.log(JSON.parse(JSON.stringify(parsed)));
 }
+
+export async function activateAndClaim3rdPartyRewardsForUser(userName: string) {
+  await setupApi();
+  await setupUsers();
+  const mangata = await getMangataInstance();
+  const keyring = new Keyring({ type: "sr25519" });
+  const testUser = new User(keyring, userName);
+  const sudo = new User(keyring, getEnvironmentRequiredVars().sudo);
+  const [newToken, newToken2] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [Assets.DEFAULT_AMOUNT, Assets.DEFAULT_AMOUNT],
+    sudo,
+    true,
+  );
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintToken(newToken2, testUser, Assets.DEFAULT_AMOUNT.muln(40e6)),
+    Assets.mintNative(testUser, Assets.DEFAULT_AMOUNT.muln(40e6).muln(2)),
+    Assets.mintToken(newToken, sudo, Assets.DEFAULT_AMOUNT.muln(40e6)),
+    Assets.mintNative(sudo, Assets.DEFAULT_AMOUNT.muln(40e6).muln(2)),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.muln(1e6),
+        newToken,
+        Assets.DEFAULT_AMOUNT.muln(1e6),
+      ),
+    ),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT,
+        newToken2,
+        Assets.DEFAULT_AMOUNT,
+      ),
+    ),
+  );
+  const liqId = await getLiquidityAssetId(MGA_ASSET_ID, newToken2);
+  await promotePool(sudo.keyRingPair, liqId, 20);
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(
+      sudo,
+      await ProofOfStake.rewardPool(
+        newToken2,
+        MGA_ASSET_ID,
+        newToken,
+        Assets.DEFAULT_AMOUNT.muln(1e6),
+        2,
+      ),
+    ),
+    Sudo.sudoAs(
+      testUser,
+      await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+        liqId,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        newToken,
+      ),
+    ),
+  );
+  await waitForRewards(testUser, liqId, 30, newToken);
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(
+      testUser,
+      await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
+    ),
+  );
+  const userRewardsTokenBalance = (
+    await mangata.query.getTokenBalance(
+      newToken.toString(),
+      testUser.keyRingPair.address,
+    )
+  ).free;
+
+  return {
```

Comment:
you do not need to return anything, plz remove it and print some relevant info like: 
testLog.getLog().info( `Setup rewards for token-ID {tokenId} and user {userAddress} , for until session: {sessionNumber}` )

Reply:
Done